examples/AffinePlaneCurve.jl

```diff
@@ -200,13 +205,14 @@ end
 
 ################################################################################
 # Check reducedness by computing a factorization
+# TODO: change for a direct squarefree check when available.
 
 @doc Markdown.doc"""
 isreduced(C::AffinePlaneCurve)
-> Given an affine plane curve C, returns true if the curve is reduced, and false otherwise.
+> Returns true if C is reduced, and false otherwise.
 """
 function Oscar.isreduced(C::AffinePlaneCurve)
-  if C.components == Dict()
+  if isempty(C.components)
      D = factor(C.eq)
      C.components = Dict(AffinePlaneCurve(x) => D.fac[x] for x in keys(D.fac))
   end
```

Comment:
Below this line, I see this code: `[C.components[x] for x in keys(C.components)] == [1 for x in keys(C.components)]`. And around line 199, I see; `collect(values(C.components)) != [1]`.

More efficient would be to use `all`,  e.g. I think this might work: `all(x -> x = 1, values(C.components))` (and perhaps one can do even better, but this should already be an improvement, I guess?)